### ISSUE TITLE
code-block default language

### ISSUE THREAD

**Issue by andreacassioli** [sphinx-doc/sphinx#1851]
Hi,
it would be useful and smart to let the language in code-block to be the global one by default. For instance if I write

```
.. highlight:: c

.. code-block::

   int a;
```

I would expect the code block highlighting to be in c. Does it make sense?

**Comment by andreacassioli** [comment]
Maybe the cleanest way is to get the highlight from the default and then let the user override:

```
.. code-block::
   :lang: c

   my_func(double* c);
```

Wouldn't be OK?

**Comment by tk0miya** [comment]
+1: reasonable.

**Comment by tk0miya** [comment]
Fixed by #6015.
Thank you for your proposal!
